numpy/lib/_ufunclike_impl.py

```diff
@@ -57,7 +57,7 @@ def fix(x, out=None):
 
     """
     # promote back to an array if flattened
-    res = nx.asanyarray(nx.ceil(x, out=out))
+    res = nx.asanyarray(nx.ceil(x, out=out if out is not x else None))
```

Comment:
This will now allocate a new array, which is typically something to be avoided when using `out`. What about
```
res = nx.asanyarray(nx.ceil(x, out=out, where=np.less(x, 0))
```